### PR TITLE
[Backend] Build Lightning-Kokkos against main; Extend support for expval and var of all observables

### PR DESCRIPTION
**Pull request by maliasadi**
As the result of PR [#47](https://github.com/PennyLaneAI/pennylane-lightning-kokkos/pull/47) in Lightning-Kokkos.

In this PR,
- Adding support for computing the variance of all observables.
- Updating `expval` by using the native Kokkos kernel in the simulator.
- Building Lightning-Kokkos against the main branch instead of the latest version in both the github action and development mode.

### CODE REVIEW

**Review by dime10 — COMMENTED** [review]
> As the result of PR https://github.com/PennyLaneAI/pennylane-lightning-kokkos/pull/47 in Lightning-Kokkos.

I don't understand this PR description 😅 

Looking at the code am I right to think variance is now supported on kokkos for all observables?

The other thing this PR does it fix the CI builds of kokkos on main by switching development to  target main correct?

File: runtime/README.rst [hunk 1]
```diff
@@ -70,7 +70,7 @@ The following table shows the available devices along with supported features:
      - All observables; Finite-shots not supported
    * - Variance
      - Only for ``Identity``, ``PauliX``, ``PauliY``, ``PauliZ``, and ``Hadamard``; Finite-shots not supported
-     - Not supported
+     - All observables; Finite-shots not supported
```
Comment (dime10) [inline runtime/README.rst:73]: So lightning.kokkos has overtaken lightning.cpu in feature support (if cpu doesn't support variance on all observables)? :D

**Review by dime10 — APPROVED** [review]
🚀

File: runtime/README.rst (hunk 1 above)
Comment (maliasadi) [reply runtime/README.rst:73]: Even, Lightning has now support for all observables if we build it against the master branch :D I'll update that support within the next release of Lightning ;)

**Pull request merged**